app/components/post_draft/draft_input/index.tsx

```diff
@@ -133,16 +133,102 @@ export default function DraftInput({
     const intl = useIntl();
     const serverUrl = useServerUrl();
     const theme = useTheme();
+    const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState(false);
+    const [isEmojiPickerFocused, setIsEmojiPickerFocused] = useState(false);
 
     const handleLayout = useCallback((e: LayoutChangeEvent) => {
         updatePostInputTop(e.nativeEvent.layout.height);
     }, []);
 
+    // To keep track of cursor position and to retrieve the latest value of cursor position
+    const cursorPositionRef = useRef(cursorPosition);
+    useEffect(() => {
+        cursorPositionRef.current = cursorPosition;
+    }, [cursorPosition]);
+
+    const handleEmojiPress = (emojiName: string) => {
```

Comment:
This is passed as a prop, so please use a `useCallback`.

Reply:
Done.